dapps/marketplace/src/components/TelegramLoginButton.js

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect, memo } from 'react'
+
+const TelegramLoginButton = ({ redirectURL, buttonText, className }) => {
+  let rootEl
+
+  useEffect(() => {
+    const script = document.createElement('script')
+    script.async = true
+    script.setAttribute('src', 'https://telegram.org/js/telegram-widget.js?7')
+    script.setAttribute(
+      'data-telegram-login',
+      process.env.TELEGRAM_BOT_USERNAME
+    )
+    script.setAttribute('data-auth-url', redirectURL)
+    script.setAttribute('data-userpic', false)
+    script.setAttribute('data-size', 'large')
+
+    rootEl.appendChild(script)
+  }, [redirectURL])
+
+  return (
+    <div ref={el => (rootEl = el)} className="telegram-login-button">
+      <button className={className}>{buttonText}</button>
```

Comment:
lets detect an `onClick` here since if that function gets called user has most probably miss configured the telegram bot. We can display some sort of a warning about that

Reply:
The iframe still renders on top of the button though. So onClick will never be called.

Reply:
aha I see... nvm then :)